### ISSUE TITLE
.github/workflows/windows-build.yml failing in master

### ISSUE THREAD

**Issue by justinvp** [pulumi/pulumi#4108]
For example:

https://github.com/pulumi/pulumi/runs/517824538

https://pipelines.actions.githubusercontent.com/I4JhI7AD9DIK0DK9Dg4gHSXXNoWQh7TNpUrWvWXAHzW5wLBXtg/_apis/pipelines/1/runs/255/signedlogcontent/3?urlExpires=2020-03-18T23%3A04%3A53.9199346Z&urlSigningMethod=HMACV1&urlSignature=NaLm8uJ%2FnjDFwqbcbfuS0KKoFyOAa2z8P78sHNQe%2FfM%3D

```
2020-03-18T22:39:11.9738546Z          [                   empty\python ] Creating a virtualenv for this projectÃ 
2020-03-18T22:39:11.9741078Z          [                   empty\python ] Pipfile: C:\Users\RUNNER~1\AppData\Local\Temp\p-it-fv-az104-python-1cab53d7-880402991\Pipfile
2020-03-18T22:39:11.9745033Z          [                   empty\python ] Using C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe (3.7.6) to create virtualenvÃ 
2020-03-18T22:39:11.9748191Z          [                   empty\python ] 
2020-03-18T22:39:11.9757648Z          [    ] Creating virtual environment...
2020-03-18T22:39:11.9761542Z          [=   ] Creating virtual environment...
2020-03-18T22:39:11.9764396Z          [==  ] Creating virtual environment...
2020-03-18T22:39:11.9766520Z          [=== ] Creating virtual environment...
2020-03-18T22:39:11.9769638Z          [ ===] Creating virtual environment...
2020-03-18T22:39:11.9778639Z          [  ==] Creating virtual environment...
2020-03-18T22:39:11.9782785Z          [   =] Creating virtual environment...
2020-03-18T22:39:11.9785293Z          [    ] Creating virtual environment...
2020-03-18T22:39:11.9831279Z          [   =] Creating virtual environment...
2020-03-18T22:39:11.9848230Z          [  ==] Creating virtual environment...
2020-03-18T22:39:11.9874565Z          [ ===] Creating virtual environment...
2020-03-18T22:39:11.9896976Z          [====] Creating virtual environment...
2020-03-18T22:39:11.9948797Z          [=== ] Creating virtual environment...
2020-03-18T22:39:11.9958216Z          [==  ] Creating virtual environment...
2020-03-18T22:39:11.9968390Z          [=   ] Creating virtual environment...
2020-03-18T22:39:11.9976777Z          [    ] Creating virtual environment...
2020-03-18T22:39:11.9984509Z          [=   ] Creating virtual environment...
2020-03-18T22:39:11.9988890Z          [==  ] Creating virtual environment...
2020-03-18T22:39:11.9990318Z          [=== ] Creating virtual environment...
2020-03-18T22:39:11.9994610Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.0010674Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.0029416Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.0034662Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.0050860Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.0097573Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.0111215Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.0118270Z          [====] Creating virtual environment...
2020-03-18T22:39:12.0148450Z          [=== ] Creating virtual environment...RuntimeError: failed to find interpreter for Builtin discover of python_spec='C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe'
2020-03-18T22:39:12.0158558Z          [                   empty\python ] 
2020-03-18T22:39:12.0163205Z          [                   empty\python ] Failed creating virtual environment 
2020-03-18T22:39:12.0165062Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\vendor\click\decorators.py", line 17, in new_func
2020-03-18T22:39:12.0288437Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:       return f(get_current_context(), *args, **kwargs)
2020-03-18T22:39:12.0303832Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\cli\command.py", line 208, in cli
2020-03-18T22:39:12.0305115Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:       clear=state.clear,
2020-03-18T22:39:12.0306967Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 574, in ensure_project
2020-03-18T22:39:12.0382168Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:       pypi_mirror=pypi_mirror,
2020-03-18T22:39:12.0441954Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 506, in ensure_virtualenv
2020-03-18T22:39:12.0870832Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:       python=python, site_packages=site_packages, pypi_mirror=pypi_mirror
2020-03-18T22:39:12.0871608Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 935, in do_create_virtualenv
2020-03-18T22:39:12.0885844Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]:       extra=[crayons.blue("{0}".format(c.err)),]
2020-03-18T22:39:12.0965754Z          [                   empty\python ] [pipenv.exceptions.VirtualenvCreationException]: 
2020-03-18T22:39:12.0985674Z          [                   empty\python ] Failed to create virtual environment.
2020-03-18T22:39:12.1088279Z          [                   empty\python ] 
2020-03-18T22:39:12.1103564Z          [           stack_outputs\python ] Invoke 'C:\hostedtoolcache\windows\Python\3.7.6\x64\Scripts\pipenv.exe --python 3.7' failed: exit status 1
2020-03-18T22:39:12.1113186Z          [           stack_outputs\python ] Creating a virtualenv for this projectÃ 
2020-03-18T22:39:12.1119240Z          [           stack_outputs\python ] Pipfile: C:\Users\RUNNER~1\AppData\Local\Temp\p-it-fv-az104-python-2e07e4d2-832398358\Pipfile
2020-03-18T22:39:12.1147120Z          [           stack_outputs\python ] Using C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe (3.7.6) to create virtualenvÃ 
2020-03-18T22:39:12.1149980Z          [           stack_outputs\python ] 
2020-03-18T22:39:12.1171908Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1181366Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1186639Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1209985Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1285058Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1298398Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1371696Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1383093Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1387822Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1391566Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1395526Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1424457Z          [====] Creating virtual environment...
2020-03-18T22:39:12.1492030Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1498178Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1500650Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1509323Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1514619Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1516855Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1526414Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1529740Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1531529Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1535989Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1541748Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1545978Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1551392Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1556455Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1557178Z          [====] Creating virtual environment...
2020-03-18T22:39:12.1559234Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1561004Z          [==  ] Creating virtual environment...RuntimeError: failed to find interpreter for Builtin discover of python_spec='C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe'
2020-03-18T22:39:12.1564078Z          [           stack_outputs\python ] 
2020-03-18T22:39:12.1567442Z          [           stack_outputs\python ] Failed creating virtual environment 
2020-03-18T22:39:12.1568053Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\vendor\click\decorators.py", line 17, in new_func
2020-03-18T22:39:12.1568659Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:       return f(get_current_context(), *args, **kwargs)
2020-03-18T22:39:12.1569202Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\cli\command.py", line 208, in cli
2020-03-18T22:39:12.1569952Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:       clear=state.clear,
2020-03-18T22:39:12.1570556Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 574, in ensure_project
2020-03-18T22:39:12.1571275Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:       pypi_mirror=pypi_mirror,
2020-03-18T22:39:12.1571890Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 506, in ensure_virtualenv
2020-03-18T22:39:12.1572467Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:       python=python, site_packages=site_packages, pypi_mirror=pypi_mirror
2020-03-18T22:39:12.1573008Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 935, in do_create_virtualenv
2020-03-18T22:39:12.1573532Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]:       extra=[crayons.blue("{0}".format(c.err)),]
2020-03-18T22:39:12.1574100Z          [           stack_outputs\python ] [pipenv.exceptions.VirtualenvCreationException]: 
2020-03-18T22:39:12.1574480Z          [           stack_outputs\python ] Failed to create virtual environment.
2020-03-18T22:39:12.1574762Z          [           stack_outputs\python ] 
2020-03-18T22:39:12.1575292Z          [                 dynamic\python ] Invoke 'C:\hostedtoolcache\windows\Python\3.7.6\x64\Scripts\pipenv.exe --python 3.7' failed: exit status 1
2020-03-18T22:39:12.1575685Z          [                 dynamic\python ] Creating a virtualenv for this projectÃ 
2020-03-18T22:39:12.1576133Z          [                 dynamic\python ] Pipfile: C:\Users\RUNNER~1\AppData\Local\Temp\p-it-fv-az104-python-4c12cfe9-920456761\Pipfile
2020-03-18T22:39:12.1576521Z          [                 dynamic\python ] Using C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe (3.7.6) to create virtualenvÃ 
2020-03-18T22:39:12.1576799Z          [                 dynamic\python ] 
2020-03-18T22:39:12.1577292Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1577590Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1577851Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1578108Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1578480Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1578937Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1579207Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1579475Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1579744Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1580195Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1580469Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1580764Z          [====] Creating virtual environment...
2020-03-18T22:39:12.1581047Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1581323Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1581753Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1582023Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1582287Z          [=   ] Creating virtual environment...
2020-03-18T22:39:12.1582582Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1582848Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1583272Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1583528Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1583783Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1584308Z          [    ] Creating virtual environment...
2020-03-18T22:39:12.1584579Z          [   =] Creating virtual environment...
2020-03-18T22:39:12.1585077Z          [  ==] Creating virtual environment...
2020-03-18T22:39:12.1585358Z          [ ===] Creating virtual environment...
2020-03-18T22:39:12.1585633Z          [====] Creating virtual environment...
2020-03-18T22:39:12.1585906Z          [=== ] Creating virtual environment...
2020-03-18T22:39:12.1586206Z          [==  ] Creating virtual environment...
2020-03-18T22:39:12.1586704Z          [=   ] Creating virtual environment...RuntimeError: failed to find interpreter for Builtin discover of python_spec='C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe'
2020-03-18T22:39:12.1587017Z          [                 dynamic\python ] 
2020-03-18T22:39:12.1587570Z          [                 dynamic\python ] Failed creating virtual environment 
2020-03-18T22:39:12.1588666Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\vendor\click\decorators.py", line 17, in new_func
2020-03-18T22:39:12.1589574Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:       return f(get_current_context(), *args, **kwargs)
2020-03-18T22:39:12.1590327Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\cli\command.py", line 208, in cli
2020-03-18T22:39:12.1590961Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:       clear=state.clear,
2020-03-18T22:39:12.1591710Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 574, in ensure_project
2020-03-18T22:39:12.1592165Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:       pypi_mirror=pypi_mirror,
2020-03-18T22:39:12.1592827Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 506, in ensure_virtualenv
2020-03-18T22:39:12.1593391Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:       python=python, site_packages=site_packages, pypi_mirror=pypi_mirror
2020-03-18T22:39:12.1594076Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:   File "c:\hostedtoolcache\windows\python\3.7.6\x64\lib\site-packages\pipenv\core.py", line 935, in do_create_virtualenv
2020-03-18T22:39:12.1595077Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]:       extra=[crayons.blue("{0}".format(c.err)),]
2020-03-18T22:39:12.1595476Z          [                 dynamic\python ] [pipenv.exceptions.VirtualenvCreationException]: 
2020-03-18T22:39:12.1595835Z          [                 dynamic\python ] Failed to create virtual environment.
2020-03-18T22:39:12.1596138Z          [                 dynamic\python ] 
2020-03-18T22:39:12.2630643Z          [                   empty\python ] Wrote output to C:\Users\RUNNER~1\AppData\Local\Temp\p-it-fv-az104-python-1cab53d7-880402991\command-output\pipenv-new.20200318-223912.9e19c.log
2020-03-18T22:39:12.2638389Z          --- FAIL: TestEmptyPython (26.41s)
```

**Comment by lukehoban** [comment]
Looks like there was a regression from virtualenv `v20.0.10` to `v20.0.11` that impacts either GHA Windows builders.  See https://github.com/pypa/virtualenv/issues/1732.

**Comment by gaborbernat** [comment]
Can anyone replicate this outside of the Github workflows, or just have a creation with more verbose output. The issue seems to be:

```
RuntimeError: failed to find interpreter for Builtin discover of python_spec='C:/hostedtoolcache/windows/Python/3.7.6/x64/python.exe'
```
but it's not clear why it fails, what paths were exhausted for it, that would help to pinpoint the issue.

**Comment by gaborbernat** [comment]
Should be fixed with 20.0.12.

**Comment by lukehoban** [comment]
Thanks @gaborbernat for both the fix and the update!

These tests are passing again now (confirmed with re-run of https://github.com/pulumi/pulumi/runs/519574909?check_suite_focus=true).
